src/backend/InvenTree/plugin/registry.py

```diff
@@ -317,14 +318,23 @@ def reload_plugins(
             full_reload (bool, optional): Reload everything - including plugin mechanism. Defaults to False.
             force_reload (bool, optional): Also reload base apps. Defaults to False.
             collect (bool, optional): Collect plugins before reloading. Defaults to False.
+            clear_errors (bool, optional): Clear any previous loading errors. Defaults to False.
             _internal (list, optional): Internal apps to reload (used for testing). Defaults to None
         """
         # Do not reload when currently loading
         if self.is_loading:
             logger.debug('Skipping reload - plugin registry is currently loading')
             return
 
-        if self.loading_lock.acquire(blocking=False):
+        if not self.loading_lock.acquire(blocking=False):
+            logger.exception('Could not acquire lock for reload_plugins')
+            return
+
+        # Reset the loading error state
+        if clear_errors:
+            self.errors = {}
+
+        try:
```

Comment:
There is no coverage on these new guards that seem pretty vital. Any idea how we could cover them in a test? We are slowly but steadily losing coverage on the plugin mechanisms.